js/views/onboardingModal.js

```diff
@@ -351,6 +351,10 @@ module.exports = baseModal.extend({
                 console.log(errorThrown);
               }
             });
+          } else {
```

Comment:
What reason for failure does the Settings api return?

Have you been able to reproduce this scenario and follow through with the experience the bug fix gives you? In other words, when you show the connect modal, what happens when the user reconnects to the same server (which is what I think I would do)? I just want to make sure they don't then have to go through onboarding again, just to have the same situation occur. In that case, I think a ugly JS alert informing them that settings failed would be better. That way it would at least be clear to us that this scenario was triggered if they ever logged a bug. Either way, it sounds like a real edge case.